keps/sig-storage/1472/README.md

```diff
@@ -0,0 +1,1416 @@
+
+### Feature enablement and rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [X] Feature gate
+    - Feature gate name: CSIStorageCapacity
+    - Components depending on the feature gate:
+      - apiserver
+      - kube-scheduler
+
+* **Does enabling the feature change any default behavior?**
+  No changes for existing storage driver deployments.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we rollback
+  the enablement)?**
+  Yes.
+
+  When disabling support in the API server, the new object type and
+  thus all objects will disappear together with the new flag in
+  `CSIDriver`, which will then cause kube-scheduler to revert to
+  provisioning without capacity information. However, the new objects
+  will still remain in the etcd database.
+
+  When disabling it in the scheduler, provisioning directly reverts
+  to the previous behavior.
+
+  External-provisioner will be unable to update objects: this needs to
+  be treated with exponential backoff just like other communication
+  issues with the API server.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+
+  Stale objects will either get garbage collected via their ownership relationship
+  or get updated by external-provisioner. Scheduling with capacity information
+  resumes.
+
+* **Are there any tests for feature enablement/disablement?**
+  The e2e framework does not currently support enabling and disabling feature
+  gates. However, unit tests in each component dealing with managing data created
+  with and without the feature are necessary. At the very least, think about
+  conversion tests if API types are being modified.
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  Try to be as paranoid as possible - e.g. what if some components will restart
+  in the middle of rollout?
+
+* **What specific metrics should inform a rollback?**
+
+* **Were upgrade and rollback tested? Was upgrade->downgrade->upgrade path tested?**
+  Describe manual testing that was done and the outcomes.
+  Longer term, we may want to require automated upgrade/rollback tests, but we
+  are missing a bunch of machinery and tooling and do that now.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features,
+  APIs, fields of API types, flags, etc.?**
+  Even if applying deprecation policies, they may still surprise some users.
+
+### Monitoring requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can an operator determine if the feature is in use by workloads?**
+  Ideally, this should be a metrics. Operations against Kubernetes API (e.g.
+  checking if there are objects with field X set) may be last resort. Avoid
+  logs or events for this purpose.
```

Comment:
I think they can find out this feature is being used by querying CSIStorageCapacity objects?